src/gpu/intel/ocl/ocl_gpu_engine.cpp

```diff
@@ -153,26 +153,29 @@ status_t create_ocl_kernel_from_cache_blob(const ocl_gpu_engine_t *ocl_engine,
 cl_int maybe_print_debug_info(
         cl_int err_, cl_program program, cl_device_id dev) {
     // Return error code if verbose is not enabled.
-    if (!get_verbose(verbose_t::error) && !get_verbose(verbose_t::warn))
-        return err_;
+#if defined(DISABLE_VERBOSE)
+    return err_;
+#endif
+
+    bool is_err = err_ == CL_SUCCESS;
+    bool is_vwarn_enabled = get_verbose(verbose_t::warn);
+
+    if (is_err || !is_vwarn_enabled) return err_;
```

Comment:
```suggestion
    bool is_err = err_ != CL_SUCCESS;
    bool is_vwarn_enabled = get_verbose(verbose_t::warn);

    if (!is_err || !is_vwarn_enabled) return err_;
```
I think the use of `is_err` on L169 is correct after this change.

Reply:
You are right. Thanks for the catch!